code/game/objects/items/devices/megaphone.dm

```diff
@@ -8,14 +8,21 @@
 	righthand_file = 'icons/mob/inhands/misc/megaphone_righthand.dmi'
 	w_class = WEIGHT_CLASS_SMALL
 	siemens_coefficient = 1
+	/// When can we next play 'megaphone.ogg'? Note that this DOESN'T affect the text embiggening, just the special megaphone noise that plays when you speak into a megaphone.
 	var/spamcheck = 0
 	var/list/voicespan = list(SPAN_COMMAND)
 
 /obj/item/megaphone/suicide_act(mob/living/carbon/user)
-	user.visible_message("<span class='suicide'>[user] is uttering [user.p_their()] last words into \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
-	spamcheck = 0//so they dont have to worry about recharging
-	user.say("AAAAAAAAAAAARGHHHHH", forced="megaphone suicide")//he must have died while coding this
-	return OXYLOSS
+	user.visible_message("<span class='suicide'>[user] is using \the [src] to amplify a very special message! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	spamcheck = 0 //screw it, you're committing suicide anyway, you can get your special megaphone noise despite it being on cooldown
+	user.say(";ONE DAY WHILE ANDY WAS-", forced="megaphone suicide") //WAIT NO DON'T
+	addtimer(CALLBACK(src, .proc/manual_suicide, user), 20) //we'll give you 2 seconds to contemplate your mistake
```

Comment:
Won't this allow a wizard to mindswap with someone right after doing this?

Reply:
no, because it returns MANUAL_SUICIDE